### PR TITLE
[HMA] Make threatexchange token optional and allow updates from the UI

### PR DESCRIPTION
**Pull request by schatten**
Summary
---------
### [`a962d715`](https://github.com/facebook/ThreatExchange/pull/898/commits/a962d7158441d6983fccb30f42cd407f949851a6) Add an API to update the te_api_token value, also make it an optional tfvar
- includes webapp API method too  
### [`12a665e5`](https://github.com/facebook/ThreatExchange/pull/898/commits/12a665e5d9156f0a89a33cbcb4a5bf4934da7e98) Allow updating TE Token from ThreatExchange settings page
- includes empty state for when privacy groups have not been synced. - removes card border around the test dataset because it looked weird as we added more settings - Uses <SettingsTabPane> at par with most other settings pages  

Test Plan
---------
* Cleared all ThreatExchangeConfig manually to see empty state.
Screen Shot 2021-12-19 at 11 08 43

* Used the token editor to supply a malformed token
Screen Shot 2021-12-20 at 09 20 49

* Used the token to supply a well-formed but invalid token

Screen Shot 2021-12-20 at 09 20 59

* Upon submitting this well-formed but invalid token
Screen Shot 2021-12-20 at 09 21 06

* Now, submit a valid token (not showing the token for security)
Screen Shot 2021-12-20 at 09 21 34

* Used the sync button to get privacy groups back
Screen Shot 2021-12-20 at 09 23 13

### CODE REVIEW

**Review by BarrettOlson — APPROVED** [review]
LGTM! 
Can you confirm the "Create" button still works for the built-in test dataset, before merge?

nit/suggestion: have `@datasets_api.post("/update-threatexchange-token")` accept empty string OR add a `@datasets_api.delete("/update-threatexchange-token")` endpoint to allow for the removal of the access token from the system.

File: hasher-matcher-actioner/hmalib/lambdas/api/datasets.py [hunk 1]
```diff
@@ -434,4 +436,26 @@ def delete_match_settings(
             )
         return bottle.abort(400, f"No match_settings for pg_id {key} found")
 
+    @datasets_api.post("/update-threatexchange-token")
```
[the remark this answers is not in the document]
Comment (schatten) [reply hasher-matcher-actioner/hmalib/lambdas/api/datasets.py:439]: I do not feel like we have used HTTP verb + URL meaningfully like REST. We are doing plain-old RPC with an optional JSON payload.

My rationale right now is to decide between GET and POST based on:
* whether we are okay with the browser or any part of the internet accidentally cache response (go with GET). So for 'actions' like create / delete / remove / edit go with POST.
* Is there a request payload (go with POST) even if a Fetch kind of a requst.

**Comment by schatten** [comment]
> LGTM! Can you confirm the "Create" button still works for the built-in test dataset, before merge?

Done.

> nit/suggestion: have `@datasets_api.post("/update-threatexchange-token")` accept empty string OR add a `@datasets_api.delete("/update-threatexchange-token")` endpoint to allow for the removal of the access token from the system.

Can you give me a use-case for removing the token?

**Comment by BarrettOlson** [comment]
> > LGTM! Can you confirm the "Create" button still works for the built-in test dataset, before merge?
> 
> Done.
> 
> > nit/suggestion: have `@datasets_api.post("/update-threatexchange-token")` accept empty string OR add a `@datasets_api.delete("/update-threatexchange-token")` endpoint to allow for the removal of the access token from the system.
> 
> Can you give me a use-case for removing the token?

I no longer want to use live data/be connected, I started using TE data before I meant to do so, OR just as yet another kill switch to have in case the others don't work as expected

**Pull request merged**